typhon/plugins/core.py

```diff
@@ -101,13 +104,70 @@ def send_new_value(self, value=None, **kwargs):
             dtype = self.signal.describe()[self.signal.name]['dtype']
             # Only way this raises a KeyError is if ophyd is confused
             self.signal_type = _type_map[dtype][0]
+
         try:
-            value = self.signal_type(value)
+            if self.signal_type is not np.ndarray:
+                value = self.signal_type(value)
             self.new_value_signal[self.signal_type].emit(value)
         except Exception:
             logger.exception("Unable to update %r with value %r.",
                              self.signal.name, value)
 
+        if severity is not None:
+            try:
+                self.new_severity_signal.emit(severity)
+            except Exception:
+                logger.exception("Unable to update %r severity with value %r.",
+                                 self.signal.name, severity)
+
+    def send_metadata(self, **kwargs):
+        """
+        Update the UI with new Signal metadata
+        """
+
+        if self.signal.connected:
+            info = self.signal.describe()[self.signal.name]
+            # Shove in connected, read_access, write_access into the desc to
+            # make my generic loop simpler, thereby confusing everyone else
+            info.update(**kwargs)
+        else:
+            # We only have legitimate access to what's given in the metadata
+            # callback, as the signal isn't connected yet :(
+            info = kwargs
+
+        key_to_signal = {
```

Comment:
I have a feeling this dictionary might be generally useful. Thoughts about putting it as a class attribute?